internal/publish/publish.go

```diff
@@ -270,18 +265,28 @@ func (p *defaultPublisher) doPublish() error {
 
 	p.setContentInfo(p.serverPublisher.GetContentInfo(contentID))
 
+	manifest, err := p.createManifest()
+	if err != nil {
+		return err
+	}
+
 	err = p.serverPublisher.PreFlightChecks()
 	if err != nil {
 		return err
 	}
 
-	bundleFile, err := p.createBundle()
+	bundleFile, err := p.createBundle(manifest)
 	if err != nil {
 		return err
 	}
 	defer bundleFile.Close()
 	defer os.Remove(bundleFile.Name())
 
+	err = p.readInterpreterDetails()
```

Comment:
I see that `p.readInterpreterDetails` is don't here and inside of `p.createBundle` above. Can we remove this if `p.createBundle` errors on this?

Reply:
Yep - just removed it; we only want to call it inside createBundle.